x-pack/plugins/alerting/server/lib/alert_instance.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { State, Context } from '../types';
+
+interface ConstructorOptions {
+  state?: Record<string, any>;
+  meta?: Record<string, any>;
+}
+
+export class AlertInstance {
+  private _shouldFire: boolean = false;
```

Comment:
Some of the variables interfere with their public functions so for consistency I prefix `_` to them all. (ex `_shouldFire` and `shouldFire()` would interfere otherwise)